shaderdb tests for GraphicsFuzz (spirv-fuzz) issues

afd opened the pull request:
Contributes shaderdb tests for issues #1034, #1035, #1036 and #1037.

amdvlk-admin commented:
Can one of the admins verify this patch?

JaxLinAMD commented:
retest this please

amdvlk-admin commented:
### Test summary for commit 94d8a91ef5dd63d50f488a81ca0b9cf0199a4d49
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 4ae736bdbc5d5dee59851ac564c5e21d807b44b0</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 87430a6fa88da11473eed49603def2fa47b51fa0</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 88613f67ca2e3e64cc1c7c4ccd47e126495e6e23</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/143766)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26276/47929 (54.8%)</li>
<li>Failed: 0/47929 (0.0%)</li>
<li>Skipped: 21653/47929 (45.2%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26275/47929 (54.8%)</li>
<li>Failed: 0/47929 (0.0%)</li>
<li>Skipped: 21654/47929 (45.2%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24807/47908 (51.8%)</li>
<li>Failed: 0/47908 (0.0%)</li>
<li>Skipped: 23101/47908 (48.2%)</li>
</ul></details>


</ul></details>

JaxLinAMD commented:
shaderdb tests failed, shall I merge it?

kuhar commented on llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpIAddCarry.spvasm:
```diff
@@ -0,0 +1,46 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc --verify-ir -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; XFAIL *
```
```suggestion
; XFAIL: *
```

kuhar commented on llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpPhiOnUndefVoid.spvasm:
```diff
@@ -0,0 +1,47 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc --verify-ir -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; XFAIL *
```
```suggestion
; XFAIL: *
```

kuhar commented on llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestOpPhiOnVoidFunctionResult.spvasm:
```diff
@@ -0,0 +1,53 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc --verify-ir -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; XFAIL *
```
```suggestion
; XFAIL: *
```

kuhar commented on llpc/test/shaderdb/fuzzer/GraphicsFuzz_TestSwitch.spvasm:
```diff
@@ -0,0 +1,87 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc --verify-ir -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; XFAIL *
```
```suggestion
; XFAIL: *
```

kuhar commented:
> shaderdb tests failed, shall I merge it?

don't merge until the tests start passing

JaxLinAMD commented:
@afd  , are you looking at the failures?

afd commented:
Sorry for the delay!  I have fixed the issues with the XFAIL syntax pointed out by @kuhar, and I also removed the two tests that involve OpPhi on void in light of our discussion about this.

JaxLinAMD commented:
retest this please

JaxLinAMD commented:
@afd , could you check if your branch is rebased onto the latest dev branch?

JaxLinAMD commented:
retest this please

amdvlk-admin commented:
### Test summary for commit 2b8643f968aea5ae3a85193cfa8b6392001a4eb7
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 76544f9b1f743ab50e1cdd8db25acc6bf27d9e8b</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev 63761ecfd2d3cfa61a756ce0a72e1ca1cdae333a</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev fde3d4e2bd90743606ca7b724bf62a53cacf4615</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/144501)</summary><ul><li>Built with version 1.2.2.2</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 26321/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21853/48174 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 26320/48174 (54.6%)</li>
<li>Failed: 0/48174 (0.0%)</li>
<li>Skipped: 21854/48174 (45.4%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 24852/48153 (51.6%)</li>
<li>Failed: 0/48153 (0.0%)</li>
<li>Skipped: 23301/48153 (48.4%)</li>
</ul></details>


</ul></details>
